### PR TITLE
fix #50: ApiFailure and ApiError impl std::error::Error

### LINKED ISSUE #50: ApiFailure should impl Error
To integrate with Failure and the rest of the Rust error ecosystem, `std::error::Error` should be implemented for `cloudflare::framework::response::apifail::ApiFailure`. Otherwise, it's hard to use `cloudflare-rs` with `?` in a function that returns `Fallible`.

### PR DESCRIPTION
**Pull request by adamchalmers**

### CODE REVIEW

File: src/framework/response/apifail.rs [hunk 1]
```diff
@@ -37,6 +38,7 @@ impl PartialEq for ApiError {
 
 impl Eq for ApiError {}
 impl Eq for ApiErrors {}
+impl Error for ApiError {}
```
Comment (nickvollmar) [inline src/framework/response/apifail.rs]: Not ApiErrors?

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "cloudflare"
-version = "0.3.1"
+version = "0.4.1"
```
Comment (sssilver) [inline Cargo.toml]: Isn't this something we only bump up during release?

File: src/framework/response/apifail.rs (hunk 1 above)
Comment (adamchalmers) [reply src/framework/response/apifail.rs]: No, ApiErrors doesn't impl Display, which is required for impling Error.

File: Cargo.toml (hunk 2 above)
Comment (adamchalmers) [reply Cargo.toml]: Good point

**Pull request merged**